product-mini/platforms/posix/main.c

```diff
@@ -287,6 +287,8 @@ moudle_destroyer(uint8 *buffer, uint32 size)
 #if WASM_ENABLE_GLOBAL_HEAP_POOL != 0
 #ifdef __NuttX__
 static char global_heap_buf[WASM_GLOBAL_HEAP_SIZE * BH_KB] = { 0 };
+#elif WASM_ENABLE_SPEC_TEST != 0
```

Comment:
How about merge these three definitions into one and define a platform dependent size (WASM_GLOBAL_HEAP_SIZE) for it ?

Reply:
Done, I defined the WASM_GLOBAL_HEAP_SIZE with unit bytes but not KB, is it OK for Nuttx?

Reply:
It's better to change https://github.com/bytecodealliance/wasm-micro-runtime/blob/22c235b5ec5c473579984a51e2bd1207f4570201/product-mini/platforms/nuttx/wamr.mk#L264 together to get correct heap pool size.

Reply:
How to change, like this?
```cmake
CFLAGS += -DWASM_GLOBAL_HEAP_SIZE=$(CONFIG_INTERPRETERS_WAMR_GLOBAL_HEAP_POOL_SIZE * 1024) 
```

Reply:
Yes, it works.